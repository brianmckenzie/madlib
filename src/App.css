.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 48px;
  font-weight: bold;
  color: #00d8ff;
}

.Madlib-text {
  color: white;
  font-size: 32px;
  font-weight: normal;
  font-family: Georgia,serif; 
}

.Form {
  font-size: 16px;
}
