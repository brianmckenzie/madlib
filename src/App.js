import React, { Component } from 'react';
import Madlib from './components/Madlib';
import './App.css';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      word1: 'dark',
      word2: 'stormy',
      word3: 'shot',
      word4: 'door',
      word5: 'The maid',
      word6: 'a pirate ship'
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">

          <p>
            Brian's Madlib Demo
          </p>

          <div className="Form">
            <form>
              
              <label>
                1:&nbsp;
                <input type="text" name="word1" id="word1" value={this.state.word1} onChange={this.handleChange} />
              </label>&nbsp;&nbsp;

              <label>
                2:&nbsp;
                <input type="text" name="word2" id="word2" value={this.state.word2}  onChange={this.handleChange} />
              </label>&nbsp;&nbsp;

              <label>
                3:&nbsp;
                <input type="text" name="word3" id="word3" value={this.state.word3}  onChange={this.handleChange} />
              </label>&nbsp;&nbsp;

              <label>
                4:&nbsp;
                <input type="text" name="word4" id="word4" value={this.state.word4}  onChange={this.handleChange} />
              </label>&nbsp;&nbsp;

              <label>
                5:&nbsp;
                <input type="text" name="word5" id="word5" value={this.state.word5}  onChange={this.handleChange} />
              </label>&nbsp;&nbsp;

              <label>
                6:&nbsp;
                <input type="text" name="word6" id="word6" value={this.state.word6}  onChange={this.handleChange} />
              </label>&nbsp;&nbsp;

            </form>
          </div>
          <br/>

          <Madlib word1={this.state.word1} 
            word2={this.state.word2} 
            word3={this.state.word3} 
            word4={this.state.word4} 
            word5={this.state.word5} 
            word6={this.state.word6}
          />

        </header>
      </div>
    );
  }
}

export default App;
