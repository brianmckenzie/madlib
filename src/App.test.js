import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Rendered app matches snapshot', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  // expect(component.state().word1).toEqual('dark');
  // component.handleChange();
  // expect(handleChange).toHaveBeenCalled();
});
