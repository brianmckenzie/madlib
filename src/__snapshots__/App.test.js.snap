// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered app matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      Brian's Madlib Demo
    </p>
    <div
      className="Form"
    >
      <form>
        <label>
          1: 
          <input
            id="word1"
            name="word1"
            onChange={[Function]}
            type="text"
            value="dark"
          />
        </label>
          
        <label>
          2: 
          <input
            id="word2"
            name="word2"
            onChange={[Function]}
            type="text"
            value="stormy"
          />
        </label>
          
        <label>
          3: 
          <input
            id="word3"
            name="word3"
            onChange={[Function]}
            type="text"
            value="shot"
          />
        </label>
          
        <label>
          4: 
          <input
            id="word4"
            name="word4"
            onChange={[Function]}
            type="text"
            value="door"
          />
        </label>
          
        <label>
          5: 
          <input
            id="word5"
            name="word5"
            onChange={[Function]}
            type="text"
            value="The maid"
          />
        </label>
          
        <label>
          6: 
          <input
            id="word6"
            name="word6"
            onChange={[Function]}
            type="text"
            value="a pirate ship"
          />
        </label>
          
      </form>
    </div>
    <br />
    <div
      className="Madlib-text"
    >
      <p>
        It was a 
        dark
         and 
        stormy
         night. Suddenly, a 
        shot
         rang out! 
        <br />
        <br />
        A 
        door
         slammed. 
        The maid
         screamed.
        <br />
        <br />
        Suddenly, 
        a pirate ship
         appeared on the horizon!
      </p>
    </div>
  </header>
</div>
`;
